src/apps/finance/hooks/useBalances.js

```diff
@@ -38,13 +35,28 @@ const useBalances = (timeout = 7000) => {
   const [tokenWithBalance, setTokenWithBalance] = useState([])
   const [loadingBalances, setLoadingBalances] = useState(true)
 
+  const vaultAddress = useRef()
+
   const chainId = 100 // TODO HANDLE CHAINS
-  const vaultContract = useContractReadOnly(
-    getAppByName(apps, VAULT_NAME).address,
-    vaultBalanceAbi,
+
+  const financeContract = useContractReadOnly(
```

Comment:
We can get the finance ABI from `connectedFinanceApp._app.abi()`. This way we can remove the `Finance.json` file